core/src/main/java/com/exadel/aem/toolkit/api/handlers/Target.java

```diff
@@ -20,123 +20,339 @@
-        if (other != null) {
-            getChildren().add(other);
-        }
-    }
 
-    default void addTarget(Target other, int position) {
-        if (other != null && other.getParent() != null) {
-            other.getParent().getChildren().remove(other);
-        }
-        if (other != null) {
-            getChildren().add(position, other);
-        }
+    /**
+     * Appends an existing {@code Target} object as a child of the current instance. If the other {@code Target} has been
+     * attached to another parent, its parent is swapped for the current instance
+     * @param other Non-null {@code Target} object
+     */
+    default void addTarget(Target other) {
+        addTarget(other, getChildren().size());
     }
 
+    /**
+     * Adds an existing {@code Target} object as a child of the current instance. If the other {@code Target} has been
+     * attached to another parent, its parent is swapped for the current instance. The new child is placed before the
+     * specified existing child
+     * @param other Non-null {@code Target} object
+     * @param near Existing child {@code Target} of the current instance
+     */
     default void addTarget(Target other, Target near) {
-        int position = near != null ? getChildren().indexOf(near) : -1;
-        if (position > -1) {
-            addTarget(other, position);
-        } else {
-            addTarget(other);
-        }
+        addTarget(other, near, false);
     }
 
+    /**
+     * Adds an existing {@code Target} object as a child of the current instance. If the other {@code Target} has been
+     * attached to another parent, its parent is swapped for the current instance. The new child is placed either before
+     * or after the specified existing child
+     * @param other Non-null {@code Target} object
+     * @param near Existing child {@code Target} of the current instance
+     * @param placeAfter True to insert the new child after the specified sibling. Otherwise, it will be inserted before
+     *                   the existing sibling
+     */
     default void addTarget(Target other, Target near, boolean placeAfter) {
-        if (!placeAfter) {
-            addTarget(other, near);
-            return;
-        }
         int position = near != null ? getChildren().indexOf(near) : -1;
-        if (position > -1 && position < getChildren().size() - 1) {
-            addTarget(other, position + 1);
-        } else {
-            addTarget(other);
+        if (placeAfter) {
+            position++;
         }
+        addTarget(other, position);
     }
 
+    /**
+     * Adds an existing {@code Target} object as a child of the current instance. If the other {@code Target} has been
+     * attached to another parent, its parent is swapped for the current instance. The new child is placed at the specified
+     * position
+     * @param other Non-null {@code Target} object
+     * @param position Zero-bases position in the list of existing children
```

Comment:
based

Reply:
Thank you for your thorough work and counsel!